### PR TITLE
🔨 (grapher) add CloseButton component

### PR DESCRIPTION
**Pull request by sophiamersmann**
[Cycle 2024.2: Entity selector](https://github.com/owid/owid-grapher/issues/3349) (prep work) | [Designs](https://www.figma.com/file/X5mOEX8zULS6qyHocUYdmh/Grapher-UI?type=design&node-id=2523%3A6266&mode=design&t=7edFp79OOjz6RENz-1)

## Summary

Adds a reusable `<CloseButton />` component that is used in a number of places:
- In modals
- In the settings popup
- In the slide-in drawer

### CODE REVIEW

**Comment by sophiamersmann** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3434?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3436** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3436?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3435** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3435?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3434** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3434?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3373** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3373?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3353** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3353?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#3356](https://github.com/owid/owid-grapher/pull/3356) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3356?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#3363](https://github.com/owid/owid-grapher/pull/3363) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3363?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sophiamersmann and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/@ourworldindata/grapher/src/modal/SourcesModal.tsx [hunk 1]
```diff
@@ -91,8 +91,8 @@ export class SourcesModal extends React.Component<
 
     @computed private get showStickyModalHeader(): boolean {
         const modalWidth = this.modalBounds.width - 2 * this.modalPadding
-        const dismissButtonWidth = 32
-        return (modalWidth - MAX_CONTENT_WIDTH) / 2 < dismissButtonWidth
+        const closeButtonWidth = 32
```
Comment (marcelgerber) [inline packages/@ourworldindata/grapher/src/modal/SourcesModal.tsx:94]: Maybe it makes sense to make this a constant inside `CloseButton.tsx` that we can import here, so it's easier to keep in sync and find it.

File: packages/@ourworldindata/grapher/src/closeButton/CloseButton.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    &.active,
+    &:active {
+        background: $active-fill;
+        border: 1px solid $active-fill;
+    }
+
+    &.active {
+        cursor: default;
+        color: $active-text;
+    }
```
Comment (marcelgerber) [inline packages/@ourworldindata/grapher/src/closeButton/CloseButton.scss:39]: I think the `.active` class name is never used anywhere - could you check and see if it's still relevant?

**Review by marcelgerber — APPROVED** [review]
Nice work, and great that you could identify & get rid of so many duplicate css styles. Love it!

File: packages/@ourworldindata/grapher/src/closeButton/CloseButton.scss (hunk 2 above)
Comment (sophiamersmann) [reply packages/@ourworldindata/grapher/src/closeButton/CloseButton.scss:39]: Removing it...

File: packages/@ourworldindata/grapher/src/modal/SourcesModal.tsx (hunk 1 above)
Comment (sophiamersmann) [reply packages/@ourworldindata/grapher/src/modal/SourcesModal.tsx:94]: Good call!

**Comment by sophiamersmann** [comment]
### Merge activity

* **May 3, 4:04 AM EDT**: @sophiamersmann started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3434).

**Pull request merged**